### DISCUSSION TITLE
How to handle two Cubes with the same base Table?

### DISCUSSION THREAD

**Discussion by MarcIrving** [atoti/atoti#583] [Q&A] [answered]
I'm working on two cubes , cube has `Table` and `cube1`  has 'Table'
`Cube` and `Cube1` have multiple columns in commun with the same exact name , when i tried to Distinct Count a value from `Cube`
it has calculated the values `Cube1` and add it to `Cube1` relative dashboard
How to point at each cube , to avoid this problem ?
`m["City_Distinct_Count"]=tt.agg.count_distinct(Table["City"])`

**Comment by bloussou** [comment, marked as answer]
Hello,
I'm not sure to understand your issue. the following code is creating measure for `cube_1` and not `cube_2` with two cubes with the same base `Table` :
```python
import atoti as tt
import pandas as pd

session = tt.Session()

table = session.read_pandas(pd.DataFrame({"column": ["a"]}), table_name="Table")

cube_1 = session.create_cube(table, name="Cube1")
cube_2 = session.create_cube(table, name="Cube2")

h1, l1, m1 = cube_1.hierarchies, cube_1.levels, cube_1.measures
h2, l2, m2 = cube_2.hierarchies, cube_2.levels, cube_2.measures

m1["distinct"] = tt.agg.count_distinct(table["column"]) # ADD MEASURE TO cube_1
```
"if I had used `m2` like : `m2["distinct"] = tt.agg.count_distinct(table["column"])` it would have added the measure to the `cube_2`"

![image](https://user-images.githubusercontent.com/32448554/167415501-250cbdcc-bea1-436a-9b66-e2fff03720c8.png)

Please use the right words so that we can help you. A [`Cube`](https://docs.atoti.io/latest/lib/atoti/atoti.cube.html#atoti.Cube) does not have `Columns`. Cubes has `Hierarchies`, `Levels` and `Measures`. A `Table` have `Columns`. You can refer to the [tutorial](https://docs.atoti.io/latest/tutorial/tutorial.html) and [documentation](https://docs.atoti.io/latest/index.html) for more definitions and concepts.

**Reply by MarcIrving** [reply]
yes thanks , it works i thought its normal to use the same h,m,l

**Reply by ShresthaUpadhyay** [reply]
Can we have two different tables in different cubes and run it under same session?

**Reply by tibdex** [reply]
Yes.
